observability/microservices.md

```diff
@@ -0,0 +1,53 @@
+# Observability in Microservices
+
+Microservices is a very popular software architecture, where the application is arranged as a collection of loosely coupled services. Some of those services can be written in different languages by different teams.
+
+## Motivations
+
+We need to consider special cases when creating a microservice architecture from the perspective of observability. We want to capture the interactions when making requests between those microservices and correlate them.
+
+Imagine we have a microservice that accesses a database to retrieve some data as part of a request. This microservice is going to be called by someone else as part of an incoming http request or an internal process being executed. What happens if a problem occurs during the retrieval of the data (or the update of the data)? How can we associate, or correlate, that this particular call failed in the destination microservice?
+
+This is a common issue. When calling other microservices, depending on the technology stack we use, we can accidentally hide errors and exceptions that might happen on the other side. If we are using a simple REST interface, the other microservice can return a 500 HTTP status code and we don't have any idea what happen inside that microservice.
+
+More important, we don't have any way to associate our Correlation Id to whatever happens inside that microservice. Therefore, is so important to have a plan in place to be able to extend your traceability and monitoring efforts, especially when using a microservice architecture.
+
+## How to extend your tracing information between microservices
```

Comment:
Does it make sense to move some of the information here to the "tracing" page?

Reply:
where is that tracing page? this is related to only where microservices is in place.

Reply:
This is the page: https://github.com/microsoft/code-with-engineering-playbook/blob/master/observability/pillars/tracing.md

Yes, I see your point about microservices. But we don't have a lot of info on that page, so maybe it is an opportunity to start adding information there?

Reply:
Now that I see the page no. In here we're explaining how to propagate the correlation between microservice, not how to implement tracing. So it's not related with tracing per se.